app/code/Magento/Sales/Setup/UpgradeSchema.php

```diff
@@ -104,4 +107,40 @@ private function addIndexBaseGrandTotal(SchemaSetupInterface $installer)
             ['base_grand_total']
         );
     }
+    
+    private function addColumnPaymentCctype(SchemaSetupInterface $installer)
+    {
+        /** @var $connection \Magento\Framework\DB\Adapter\Pdo\MySql */
+        $connection = $installer->getConnection();
+        $gridTableName = $installer->getTable('sales_order_grid');
+        $paymentTableName = $installer->getTable('sales_order_payment');
+
+        $cctypeColumns = [
+            'type' => \Magento\Framework\DB\Ddl\Table::TYPE_TEXT,
+            'length' => 32,
+            'nullable' => true,
+            'comment' => 'Payment CC Type',
+        ];
+
+        $connection->addColumn(
+            $gridTableName,
+            'payment_cctype',
+            $cctypeColumns
+        );
+        
+        
+        //Populate "payment_cctype" in sales_order_grid with values in sales_order_payments
+        $connection->query(
+            $connection->updateFromSelect(
+                $connection->select()
+                    ->join(
+                        $paymentTableName,
+                        sprintf('%s.entity_id = %s.parent_id', $gridTableName, $paymentTableName),
+                        'cc_type AS payment_cctype'
```

Comment:
Why need to denormalize instead of dynamic joins?

Reply:
I am not sure what you mean... I looked at the `sales_order_grid.payment_method` column and followed that approach. Those lines take care of copying existing values from `sales_order_payment.cc_type` to `sales_order_grid.payment_cctype`

Reply:
Why this data needs to be copied at all? How it works when there are multiple invoices associated to order with different payment methods?

Reply:
I understand your point, but I believe that your questions are valid also for `sales_order_grid.payment_method` , that is in the core and that I took inspiration from.
Data need to be copied to populate the field in the existing orders.